core/src/main/java/nl/nn/adapterframework/webcontrol/api/BrowseQueue.java

```diff
@@ -92,21 +93,29 @@ public Response putBrowseQueue(LinkedHashMap<String, Object> json) throws ApiExc
 			if(key.equalsIgnoreCase("payload")) {
 				showPayload = Boolean.parseBoolean(entry.getValue().toString());
 			}
+			if(key.equalsIgnoreCase("lookupDestination")) {
+				lookupDestination = Boolean.parseBoolean(entry.getValue().toString());
+			}
 		}
 
-		if(jmsRealm == null)
-			throw new ApiException("No realm provided");
+		if(connectionFactory == null)
+			throw new ApiException("No connection factory provided");
 		if(destination == null)
 			throw new ApiException("No destination provided");
 		if(type == null)
 			throw new ApiException("No type provided");
 
 		try {
-			JmsBrowser<javax.jms.Message> jmsBrowser = new JmsBrowser<>();
+			JmsBrowser<javax.jms.Message> jmsBrowser = getIbisContext().createBeanAutowireByName(JmsBrowser.class);
 			jmsBrowser.setName("BrowseQueueAction");
+			String jmsRealm = JmsRealmFactory.getInstance().findJmsRealm(connectionFactory);
+			if(jmsRealm == null) {
+				throw new ApiException("connection factory ["+connectionFactory+"] does not have a realm");
+			}
```

Comment:
You should not convert the connectionFactory back to a realm and use `setJmsRealm()`, but use `setConnectionFactory()` instead.